components/button/button-add.js

```diff
@@ -0,0 +1,131 @@
+import '../colors/colors.js';
+import '../icons/icon.js';
+import '../tooltip/tooltip.js';
+import { css, html, LitElement } from 'lit';
+import { FocusMixin } from '../../mixins/focus/focus-mixin.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { labelStyles } from '../typography/styles.js';
+import { LocalizeCoreElement } from '../../helpers/localize-core-element.js';
+
+/**
+ * A component for quickly adding items to a specific locaiton.
+ */
+class ButtonAdd extends FocusMixin(LocalizeCoreElement(LitElement)) {
+	static get properties() {
+		return {
+			/**
+			 * The text to be shown in the tooltip as a hint when visible-text is false
+			 * @type {string}
+			 */
+			label: { type: String },
+			/**
+			 * The text to show in the button when visible-text is true
+			 * @type {string}
+			 */
+			text: { type: String },
+			/**
+			 * When true, show the button with icon and visible text. When false, only show icon.
+			 * @type {boolean}
+			 */
+			visibleText: { type: Boolean, attribute: 'visible-text' }
+		};
+	}
+
+	static get styles() {
+		return [labelStyles, css`
```

Comment:
I didn't use `buttonStyles` here as it felt like it complicated things more than made them easier. Let me know if I should be using that.